apps/studio/components/interfaces/Storage/StorageExplorer/CustomExpiryModal.tsx

```diff
@@ -10,94 +31,141 @@ const unitMap = {
-      visible={visible}
-      alignFooter="right"
-      confirmText="Get URL"
-      onCancel={() => onClose()}
+    <Dialog
+      open={visible}
+      onOpenChange={(open) => {
+        if (!open) {
+          handleClose()
+        }
+      }}
     >
-      <Form
-        validateOnBlur
-        initialValues={{ expiresIn: '', units: 'days' }}
-        onSubmit={async (values: any, { setSubmitting }: any) => {
-          setSubmitting(true)
-          await onCopyUrl(
-            selectedFileCustomExpiry!.name,
-            values.expiresIn * unitMap[values.units as 'days' | 'weeks' | 'months' | 'years']
-          )
-          setSubmitting(false)
-          onClose()
-        }}
-        validate={(values: any) => {
-          const errors: any = {}
-          if (values.expiresIn !== '' && values.expiresIn <= 0) {
-            errors.expiresIn = 'Expiry duration cannot be less than 0'
-          }
-          return errors
-        }}
-      >
-        {({ values, isSubmitting }: { values: any; isSubmitting: boolean }) => (
-          <>
-            <Modal.Content>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Custom expiry for signed URL</DialogTitle>
+        </DialogHeader>
+        <DialogSectionSeparator />
+        <DialogSection>
+          <Form_Shadcn_ {...form}>
+            <form
+              id={formId}
+              className="flex flex-col gap-4"
+              onSubmit={form.handleSubmit(onSubmit)}
+            >
               <p className="text-sm text-foreground-light mb-2">
                 Enter the duration for which the URL will be valid for:
               </p>
-              <div className="flex items-center space-x-2">
-                <Input disabled={isSubmitting} type="number" id="expiresIn" className="w-full" />
-                <Listbox id="units" className="w-[150px]">
-                  <Listbox.Option id="days" label="days" value="days">
-                    days
-                  </Listbox.Option>
-                  <Listbox.Option id="weeks" label="weeks" value="weeks">
-                    weeks
-                  </Listbox.Option>
-                  <Listbox.Option id="months" label="months" value="months">
-                    months
-                  </Listbox.Option>
-                  <Listbox.Option id="years" label="years" value="years">
-                    years
-                  </Listbox.Option>
-                </Listbox>
+              <div className="grid grid-cols-12 col-span-12 gap-x-2 gap-y-1">
+                <FormField_Shadcn_
+                  key="expiresIn"
+                  name="expiresIn"
+                  control={form.control}
+                  render={({ field }) => (
+                    <FormControl_Shadcn_>
+                      <Input_Shadcn_
+                        aria-label="Duration"
+                        id="expiresIn"
+                        type="number"
+                        min={0}
```

Comment:
note that the `min` attribute here eliminates the need for the previous validation logic, as the input itself will prevent values below 0 from being entered